### PR TITLE
Fix #6, add more features

### LINKED ISSUE #6: 后台警告刷屏
- 服务端
Purpur-1.20.1
- Java版本
17
- 警告信息
[11:48:23 WARN]: [BoatFly] Plugin BoatFly v1.0-SNAPSHOT generated an exception while executing task 782047
java.lang.NullPointerException: null
[11:48:23 WARN]: [BoatFly] Plugin BoatFly v1.0-SNAPSHOT generated an exception while executing task 782047
java.lang.NullPointerException: null
[11:48:23 WARN]: [BoatFly] Plugin BoatFly v1.0-SNAPSHOT generated an exception while executing task 810517

### PR DESCRIPTION
**Pull request by Lumine1909**
Bug fix:
https://github.com/Wenkrangha/BoatFly/issues/6

New features:
1. Tab complete support
2. Get flight boat directly by command

some implementation changes...

### CODE REVIEW

File: src/main/java/com/wenkrang/boatfly/command/bf.java
Comment (Wenkrangha) [inline src/main/java/com/wenkrang/boatfly/command/bf.java:1]: 关于获取飞船的命令是没必要的，在创造下，右键配方书可以直接获取飞船

File: src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java [hunk 1]
```diff
@@ -11,21 +11,23 @@
 public class KickEntity implements Listener {
     @EventHandler
     public static void OnPlayer (InventoryClickEvent event) {
-        if (event.getView().getTitle().equalsIgnoreCase("飞船控制面包")) {
+        if (event.getView().getTitle().equalsIgnoreCase("飞船控制面包")/* isn't it "版"?*/) {
```
Comment (Wenkrangha) [inline src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java:14]: 本来就是面包()

File: src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java [hunk 2]
```diff
@@ -11,21 +11,23 @@
...
                         }
                         Objects.requireNonNull(event.getWhoClicked().getVehicle()).removePassenger(event.getWhoClicked().getVehicle().getPassengers().get(0));
                     }
                     event.getWhoClicked().getWorld().playEffect(event.getWhoClicked().getLocation(), Effect.CLICK1, 1, 30);
                 }
                 if (event.getRawSlot() == 24) {
                     if (Objects.requireNonNull(event.getWhoClicked().getVehicle()).getPassengers().get(1) != null) {
-                        if (event.getWhoClicked().getVehicle().getPassengers().get(0) instanceof Player) {
-                            event.getWhoClicked().getVehicle().getPassengers().get(0).addScoreboardTag("CanExit");
+                        if (event.getWhoClicked().getVehicle().getPassengers().get(1) instanceof Player) {
```
Comment (Wenkrangha) [inline src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java:28]: 很好的修复

File: src/main/resources/plugin.yml [hunk 3]
```diff
@@ -5,4 +5,7 @@ api-version: '1.16'
 commands:
   bf:
     description: Spigot版本的飞船插件的主命令
-    usage: /bf
\ No newline at end of file
+    usage: /bf
+permissions:
+  bf.getboat:
+    default: op
```
Comment (Wenkrangha) [inline src/main/resources/plugin.yml:11]: 实际上可以去掉

**Review by Wenkrangha — CHANGES_REQUESTED** [review]
我觉得部分地方需要修改

**Comment by Wenkrangha** [comment]
你的更改我会考虑的

File: src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java [hunk 4]
```diff
@@ -11,21 +11,23 @@
...
     @EventHandler
     public static void OnPlayer (InventoryClickEvent event) {
-        if (event.getView().getTitle().equalsIgnoreCase("飞船控制面包")) {
+        if (event.getView().getTitle().equalsIgnoreCase("飞船控制面包")/* isn't it "版"?*/) {
             try {
                 if (event.getRawSlot() == 23) {
                     if (Objects.requireNonNull(event.getWhoClicked().getVehicle()).getPassengers().get(0) != null) {
                         if (event.getWhoClicked().getVehicle().getPassengers().get(0) instanceof Player) {
                             event.getWhoClicked().getVehicle().getPassengers().get(0).addScoreboardTag("CanExit");
+                            ((Player) event.getWhoClicked().getVehicle().getPassengers().get(0)).closeInventory();
```
Comment (Wenkrangha) [inline src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java:20]: 有监听器监听了

**Comment by Wenkrangha** [comment]
感谢你的提交，代码已采用，你的名字已加入贡献者列表

File: src/main/java/com/wenkrang/boatfly/command/bf.java
Comment (Lumine1909) [reply src/main/java/com/wenkrang/boatfly/command/bf.java:1]: Should we add more info to the recipe book? I don't know that feature before you mentioned :(

File: src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java (hunk 1 above)
Comment (Lumine1909) [reply src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java:14]: ()

File: src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java (hunk 4 above)
Comment (Lumine1909) [reply src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java:20]: I try it on paper-1.20.2-318, but the listener seems not work correctly.

After I click this button,
image


The panel still exist
image

Comment (Lumine1909) [reply src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java:20]: image

com.wenkrang.boatfly.event.VehicleExit.java

Listener DO NOT close player's inventory when player exit the vehicle (the exit caused by kick from the panel).

Comment (Lumine1909) [reply src/main/java/com/wenkrang/boatfly/event/GUI/KickEntity.java:20]: In addition, #6 may caused by this bug, when player kick itself and click the panel without ride on boat.

**Pull request closed without merging**